chapter 9, 10 , 11

abdul-moize opened the pull request:
all work done

abdul-moize committed:
chapter 9-11 fixed flake errors

arslanashraf7 commented on src/standard_libraries/test_re.py:
Could you do the below things here:

1. Can you format the regex something like [verbose formatting](https://diveintopython3.problemsolving.io/regular-expressions.html#verbosere)
2. Can we do something, If we are to introduce case-insensitivity?
3. Can we add one more test for a regex that would match Employee Card numbers? The format of the card number is (XX-XXXXX-44-XX)  -- Here X could be any number.

arslanashraf7 commented on src/files/test_file_reading.py:
Could you also try opening this fine in binary mode `rb`?

arslanashraf7 commented on src/files/test_file_reading.py:
How about we do this file path generation with `.format`? Could you try it? You might need to update the path declaration.

arslanashraf7 commented on src/standard_libraries/test_json.py:
How about we create another test where we generate a json, something like:

`{
"name":  "Some Name",
"Age": 23,
"grades": [{
"title": "Science",
"total": 100,
"passing": 80,
"obtained": 85
},{
"title": "Math",
"total": 100,
"passing": 80,
"obtained":70
}]}
`

arslanashraf7 reviewed (changes requested):
I've added some general suggestions, Along with them can you please add some function to test how 'pass' works?

abdul-moize committed:
changed added

abdul-moize committed:
changed added 1.1

abdul-moize commented on src/standard_libraries/test_re.py:
1. i have converted the regex from raw format to verbose format
2. we can remove the 'A-Z' to make it case sensitive. i have updated the regex to only accept lower case now
3. i have added this as well

abdul-moize committed:
changed added 1.2

abdul-moize commented on src/files/test_file_reading.py:
done

abdul-moize commented on src/files/test_file_reading.py:
done

abdul-moize commented on src/standard_libraries/test_json.py:
added a test case

abdul-moize commented:
> I've added some general suggestions, Along with them can you please add some function to test how 'pass' works?

i have updated the changes and also added a use case for pass. kindly check
